solr/core/src/java/org/apache/solr/core/NodeConfig.java

```diff
@@ -455,7 +460,32 @@ private void setupSharedLib() {
       libDirs.addAll(sharedLibs);
     }
 
-    addFoldersToSharedLib(libDirs);
+    List<Path> libDirPaths = new ArrayList<>(libDirs.size());
+    for (String libDir : libDirs) {
+      libDirPaths.add(getSolrHome().resolve(libDir).toAbsolutePath());
+    }
+    return libDirPaths;
+  }
+
+  // Get the list of protected directories, that cannot have files created within them
+  public List<Path> getProtectedDirectories() {
+    List<Path> protectedDirectories = new ArrayList<>(sharedLibPaths.size() + 4);
+
+    // Start with the shared lib directories
+    protectedDirectories.addAll(sharedLibPaths);
+
+    if (getSolrInstallDir() != null) {
+      // Add in modules and tools
+      protectedDirectories.add(ModuleUtils.getModulesPath(getSolrInstallDir()).toAbsolutePath());
+      protectedDirectories.add(getSolrInstallDir().resolve("prometheus-exporter").toAbsolutePath());
+
+      // Add in server library folders
+      protectedDirectories.add(
+          getSolrInstallDir().resolve("server").resolve("lib").toAbsolutePath());
+      protectedDirectories.add(
+          getSolrInstallDir().resolve("server").resolve("solr-webapp").toAbsolutePath());
+    }
```

Comment:
maybe assert that these actually exist?  If one day we change a name or move something, the assert will help us keep this up to date.